provisioner/template.go

```diff
@@ -552,3 +558,24 @@ func stupsNATSubnets(vpcCidr string) ([]string, error) {
 	}
 	return result, nil
 }
+
+func amiID(adapter *awsAdapter, imageName, imageOwner string) (string, error) {
+	if adapter == nil || adapter.ec2Client == nil {
+		return "", fmt.Errorf("the ec2 client is not available")
+	}
+
+	filterNameName := describeImageFilterNameName
+	filterOwnerName := describeImageFilterNameOwner
+	input := ec2.DescribeImagesInput{Filters: []*ec2.Filter{
+		{Name: &filterNameName, Values: []*string{&imageName}},
+		{Name: &filterOwnerName, Values: []*string{&imageOwner}},
```

Comment:
You could also use `aws.String()` for that. You might be able to drop the temporary variables as well. There's also [StringSlice](https://godoc.org/github.com/aws/aws-sdk-go/aws#StringSlice) for the values part.